practice/rain.py

```diff
@@ -1,86 +1,214 @@
 """ Opens a Rain Text File and Mines for information. I'm using "cells" for rows, and "boxes" for columns"""
 
-def open_file():
-    """Opens text file for analysis"""
-    with open('rainspoof.txt', 'r') as table_file:
+
+from itertools import groupby
+
+
+TEST_ROWS = [['23', 'MAR', '2016', '12'], ['22', 'MAR', '2016', '1'], ['21', 'MAR', '2016', '8']]
+TEST_COLUMNS = [['23', '22', '21'], ['MAR', 'MAR', 'MAR'], ['2016', '2016', '2016'], ['12', '1', '8']]
+
+
+def open_file(file):
+    """Opens text file for analysis.
+
+    >>> open_file('rain_text/testrain.txt')
+    ['She took all my money    And my best friend    You know the story    Here it comes again']
+    """
+    with open(file, 'r') as table_file:
         lines_of_text = table_file.readlines()
     return lines_of_text
 
 
 def cut_non_data(text):
-    """ Takes in lines of text, and only returns relevant data lines.
-    >>> cut_non_data(['I can haz cheezeburger',
-    >>> '------------------------------------------------------------------------------------------------------------------\n',
-    >>> 'This line is the only one that matters'])
-    ['This is the only one that matters']
-    """
-    TABLE_LINE_CUT_OFF = '------------------------------------------------------------------------------------------------------------------\n'
-    table_start_index = text.index(TABLE_LINE_CUT_OFF) + 1
-    table_as_text_lines = text[table_start_index:]
+    """Cuts lines of text after table line.
+
+    >>> cut_non_data([[1], [2], [3], [4], [5], [6], [7], [8], [9], [10], [11], [12], [13], [14]])
+    [[12], [13], [14]]
+    """
+    table_as_text_lines = text[11:]
     return table_as_text_lines
 
 
 def split_single_line_to_single_row(text):
-    """Splits a single line of text into a row in the form of a list of strings"""
-    return text.split('    ')
+    """Splits a single line of text into a row in the form of a list of strings.
+
+    >>> split_single_line_to_single_row('one   two   three' )
+    ['one', 'two', 'three']
+    >>> split_single_line_to_single_row('one-two-three')
+    ['one', 'two', 'three']
+    """
+    simple_row = text.split('   ')
+    full_date = simple_row[0]
+    split_dates = full_date.split('-')
+    final_row = split_dates + simple_row[1:]
+    return final_row
```

Comment:
I would say the docstrings and doctests misrepresent what this function does. It takes a string, first splits by triple spaces, then takes the first token and splits by hyphens. I assume you want this to give it a full line from the input file and get a fully parsed date. Have the docstring and doctests actually describe and test all of that functionality.